openhands/runtime/utils/runtime_build.py

```diff
@@ -183,11 +184,25 @@ def get_runtime_image_repo_and_tag(base_image: str) -> tuple[str, str]:
         if ':' not in base_image:
             base_image = base_image + ':latest'
         [repo, tag] = base_image.split(':')
-        # replace '/' with '_s_' to avoid '/' in the image name
-        # while make it a valid docker image name
-        repo = repo.replace('/', '_s_')
-        od_version = _get_package_version()
-        return get_runtime_image_repo(), f'od_v{od_version}_image_{repo}_tag_{tag}'
+        oh_version = _get_package_version()
+
+        # Hash the repo if it's too long
+        if len(repo) > 32:
+            repo_hash = hashlib.md5(repo[:-24].encode()).hexdigest()[:8]
+            repo = f'{repo_hash}_{repo[-24:]}'  # Use 8 char hash + last 24 chars
+        else:
+            repo = repo.replace('/', '_s_')
+
+        new_tag = f'oh_v{oh_version}_image_{repo}_tag_{tag}'
+
+        # if it's still too long, hash the entire image name
+        if len(new_tag) > 128:
```

Comment:
We actually could use this for the other runtime, too! Saw that docker 500 error due to too long name as well, but only once thus far.

Reply:
i thought only me encountered this issue haha - happy to see that we can hit two birds with one stone 😄